custom/enikshay/reports/historical_adherence.py

```diff
@@ -240,7 +243,7 @@ def _source_is_enikshay(case):
             return case.dynamic_case_properties().get('adherence_source') in ('enikshay', '')
 
         return sorted(
-            adherence_cases, key=lambda c: (_source_is_enikshay(c), c.modified_on)
+            adherence_cases, key=lambda c: (_source_is_enikshay(c), not c.closed, c.modified_on, c.opened_on)
```

Comment:
And the `not closed` part of the key was added because the mobile version of the calendar gives preference to open cases (because of course only open cases are synced to the device).